packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1438 @@
+    String nullDescription,
+  }) {
+    if (value == null)
+      header = nullDescription;
+
+    return new _LeafDiagnosticsValue(
+      name: name,
+      header: header ?? value.toString(),
+      object: value,
+      showNull: showNull,
+      hidden: hidden,
+      showName: showName,
+    );
+  }
+
+  /// A boolean flag property that is shown as `name` if value is true and is
+  /// true and hidden otherwise.
+  factory DiagnosticsNode.flagProperty(String name, bool value) {
+    return new _LeafDiagnosticsValue(
+      name: name,
+      header: value == true ? name : '',
+      object: value,
+      hidden: value != true,
+      showName: false);
+  }
+
+  /// An object property with a `comment` describing the value appended after
+  /// the description of the `value.
+  factory DiagnosticsNode.objectPropertyWithComment(String name, Object value, {
+    @required String comment,
+  }) {
+    return new _LeafDiagnosticsValue(
+      name: name,
+      object: value,
+      header: '$value ($comment)',
+    );
+  }
+
+  /// Creates a property with a [double] value.
+  ///
+  ///  If `showNull` is true, the property is shown even if its `value` is null.
+  /// `showNull` indicates whether the property should still be shown if its
+  /// `value` is null. `hidden` specifies whether the property should be hidden
+  /// or shown. `nullDescription` provides a description of the property value
+  /// if it is null.
+  /// If specified, exactly `fractionDigits` after the decimal point are shown.
+  /// If specified, `unit` describes the unit for the value (e.g. px).
+  factory DiagnosticsNode.doubleProperty(String name, double value, {
+    bool showNull: true,
+    int fractionDigits,
+    bool hidden: false,
+    String nullDescription,
+    String unit,
+  }) {
+    String description;
+    if (value == null) {
+      description = nullDescription ?? value.toString();
+    } else {
+      description = fractionDigits != null ?
+         value.toStringAsFixed(fractionDigits) : value.toString();
+    }
+
+    if (unit != null)
+      description = '$description$unit';
+
+    return new _LeafDiagnosticsValue(
+      name: name,
+      header: description,
+      object: value,
+      showNull: showNull,
+      hidden: hidden,
+    );
+  }
+
+  /// Property with a [double] value where computing its value might throw an
+  /// exception.
+  ///
+  /// If an exception is thrown, the value is set to null and the description
+  /// indicates what exception was thrown.
+  /// If specified, exactly `fractionDigits` after the decimal point are shown.
+  factory DiagnosticsNode.unsafeDoubleProperty(
+      String name, double computeValue(), { int fractionDigits }) {
+    try {
+      return new DiagnosticsNode.doubleProperty(name, computeValue(), fractionDigits: fractionDigits);
+    } catch (e) {
+      // TODO(jacobr): track the exception thrown as part of the Diagnostics node.
+      return new DiagnosticsNode.doubleProperty(
```

Comment:
because I prefer to avoid re-implementing duplicate logic such as fractionDigits and unit. I expect `unit` should be added to this constructor as an exception to the rule that functionality should only be added as needed.